autogpt_platform/backend/schema.prisma

```diff
@@ -19,45 +19,141 @@ model User {
+  createdAt DateTime @default(now())
+  updatedAt DateTime @updatedAt
+
+  name         String
+  description  String
+  groupIconUrl String?
+
+  users      User[] @relation("GroupUsers")
+  adminUsers User[] @relation("GroupAdminUsers")
+
+  agents       Agent[]
+  Profile      Profile[]
+  StoreListing StoreListing[]
+
+  @@index([name])
+}
+
 // This model describes the Agent Graph/Flow (Multi Agent System).
-model AgentGraph {
+model Agent {
   id        String    @default(uuid())
   version   Int       @default(1)
   createdAt DateTime  @default(now())
   updatedAt DateTime? @updatedAt
 
   name        String?
   description String?
-  isActive    Boolean @default(true)
-  isTemplate  Boolean @default(false)
 
   // Link to User model
-  userId String
-  user   User   @relation(fields: [userId], references: [id], onDelete: Cascade)
+  createdByUserId String
+  createdByUser   User   @relation(fields: [createdByUserId], references: [id], onDelete: Cascade)
+
+  isGroupAgent Boolean    @default(false)
+  groupId      String?
+  group        UserGroup? @relation(fields: [groupId], references: [id])
 
   AgentNodes                  AgentNode[]
   AgentGraphExecution         AgentGraphExecution[]
   AgentGraphExecutionSchedule AgentGraphExecutionSchedule[]
 
   // All sub-graphs are defined within this 1-level depth list (even if it's a nested graph).
-  AgentSubGraphs     AgentGraph[] @relation("AgentSubGraph")
+  AgentSubGraphs     Agent[] @relation("AgentSubGraph")
   agentGraphParentId String?
-  AgentGraphParent   AgentGraph?  @relation("AgentSubGraph", fields: [agentGraphParentId, version], references: [id, version], onDelete: Cascade)
+  AgentGraphParent   Agent?  @relation("AgentSubGraph", fields: [agentGraphParentId, version], references: [id, version], onDelete: Cascade)
+
+  ConfiguredAgents    ConfiguredAgent[]
+  UserAgents          UserAgents[]
+  UserBlockCredit     UserBlockCredit[]
+  StoreListing        StoreListing[]
+  StoreListingVersion StoreListingVersion[]
 
   @@id(name: "graphVersionId", [id, version])
+  @@map("AgentGraph")
+}
+
+//////////////////////////////////////////////////////////// 
+//////////////////////////////////////////////////////////// 
+////////////////   USER SPECIFIC DATA   ////////////////////
+//////////////////////////////////////////////////////////// 
+//////////////////////////////////////////////////////////// 
+
+model ConfiguredAgent {
+  id        String   @id @default(uuid())
+  createdAt DateTime @default(now())
+  updatedAt DateTime @updatedAt
+
+  name        String
+  description String
+
+  userId String
+  user   User   @relation(fields: [userId], references: [id])
+
+  agentId      String
+  agentVersion Int
+  agent        Agent  @relation(fields: [agentId, agentVersion], references: [id, version])
+
+  Configuration AgentNodeExecutionInputOutput[] @relation("AgentNodeExecutionInput")
+  UserAgents    UserAgents[]
+
+  @@index([userId])
 }
 
+model UserAgents {
```

Comment:
A User managed list of agents that they can see in there library

Reply:
Why is this different than getting the agents that match the collection of groups the user is in?

Reply:
This is to manage what the User sees in there library